### ISSUE TITLE
Using the "window" parameter on find and seach methods.

### ISSUE THREAD

**Issue by piroxxi** [mopidy/mopidy#1245]
Hi,

I'm currently creating an Android Application communicating with a Mopidy server using an HTTP socket. I'm currently trying to work with the "search" function, using the "window" parameter :
http://www.musicpd.org/doc/protocol/database.html

I've been trying multiple ways to write it, but none seems to be working. Here is a example :
<pre>
nc 192.168.0.9 6600
search any "Test"
...... works ......
search any "Test" 4:5
ACK [2@0] {search} incorrect arguments
search any "Test" "4:5"
ACK [2@0] {search} incorrect arguments
</pre>

Am-I doing it wrong ? Can anyone help me with that ?

Thank you,
Raphaël

**Comment by jodal** [comment]
From the docs, I think it should be:

```
search any "Test" window 4:5
```

Though, I've never seen this before, so I doubt we support it yet. The docs doesn't say anything about what it does. I assume it is for pagination of the search results?

**Comment by piroxxi** [comment]
Yes, it's a pagination argument :
<pre>
search any "Test" window 4:5
ACK [2@0] {search} incorrect arguments
</pre>

Is there any other way to limit the number of returned items when doing a search ? For some search the number of items may be really big...

**Comment by jodal** [comment]
No, not in our current MPD protocol implementation. Though, most services limit themselves. E.g. the Spotify backend returns at most 200 each of artists, albums and tracks.

**Comment by piroxxi** [comment]
Ok :)

Thanks for the quick reply ! Even tho it shouldn't be a big problem for me (I should be able to find another way), do you have any idea when such a functionality could be done ?

**Comment by adamcik** [comment]
http://git.musicpd.org/cgit/master/mpd.git/tree/NEWS lists this as a feature in the unreleased 0.20 version of MPD. So we are not lagging behind just yet :-)

**Comment by piroxxi** [comment]
Yeay ! That's great ! :)

Thanks again for the impressively quick answers ! :)
